javascript.js

```diff
@@ -0,0 +1,137 @@
+// Create a tree function that should build a pine tree out of a
+// character in the Chrome dev tools console.
+//  A key that specifies the height of the pine tree.
+// The value for the height of the tree should be from user input in 
+// a <input type="text"> field in the DOM.
+// A key that specifies which character to use to build the pine tree.
+// The character to use should be from user input in a
+// <input type="text"> field in the DOM.
+const typeOfChar = document.getElementById("Char");
+const heightOfChar = document.getElementById("heightChar")
+const buttonStart = document.getElementById("startButton");
+let treeArray = [];
+
+
+function treeObject(){
+	let tree = Object.create(null); 
+	tree.character = document.getElementById("Char").value;
+	tree.height = document.getElementById("heightChar").value;
+	
+	// console.log("Did this work, function", tree);
+	return tree;
+};
+
+function treeSpaces(height, character){//This didn't need the key because buttonListen which called it has it defined already
+	treeArray.push(`${character}`);
+	// console.log("before loop", treeArray);
+	for(let i = height; i > 0; i--){
+		treeArray.unshift(" ");//Needed to be switched to unshift because it was putting spaced to the end of the array from
+		//in contrast from it's earlier position.
+		
+		// console.log("during loop", treeArray);
+		}
+		// console.log("after loop", treeArray);
+};
+
+function buildTree(treeObj){//need button's treeObj
+	
+	
+		for (let i =  treeObj.height; i > 0; i--){//needed to be the same as the parameter
+			
+			
+			console.log(treeArray.join("")); //make array string
+
+			treeArray.push(`${treeObj.character}${treeObj.character}`);
+			treeArray.shift();
+
+		}
+		//	console.log("buildTree", treeArray);
+};
+
+function buttonListen(){//Needed to have a variable hold the treeObject() result
+		// console.log("ButtonClicked");
+		let treeObj = treeObject();
+		
+		treeSpaces(treeObj.height, treeObj.character);//This allowed us to use the keys inside //called here
+		
+		
+		buildTree(treeObj);
+
+		
+};
+
+typeOfChar.addEventListener("keypress", function(event){
+	// console.log("event", event);
+	let key = event.keycode;
+	if (key == 13){ //entered value might be a string?
+		buttonListen();
+	}
+});
+heightOfChar.addEventListener("keypress", function(event){
+	let key = event.keycode;
+	if (key == 13){
+		buttonListen();
+	}
```

Comment:
Found a fix, apparently .keycode isn't supported by many browsers anymore and .which is universally supported now.